### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by Alndor**

### CODE REVIEW

**Comment by Alndor** [comment]
Fixing some bugs

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
   // rate of failing to fire torpedos [0.0, 1.0]
   private double FAILURE_RATE = 0.0; //NOSONAR
-
+ private Random generator = new Random();
```
Comment (gurubib) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:14]: Kivalo

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +30,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (gurubib) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:33]: Zsir

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +30,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (gurubib) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:43]: Patent

**Review by gurubib — APPROVED** [review]
Kiraly

**Pull request merged**